pkg/lib/analytic/count_collector.go

```diff
@@ -84,5 +101,145 @@ func (c *CountCollector) CollectDailyCountForApp(appID string, date time.Time, n
 		return
 	}
 
+	// Signup count
+	err = c.AuditDBHandle.ReadOnly(func() error {
+		signupCountResult, err := c.QuerySignupCount(appID, &date, &nextDay)
+		if err != nil {
+			err = fmt.Errorf("failed to calculate signup count: %w", err)
+			return err
+		}
+		if signupCountResult.TotalCount == 0 {
+			// no new signup for the app, skip the signup count
+			return nil
+		}
+
+		counts = append(counts, NewCount(
+			appID,
+			signupCountResult.TotalCount,
+			date,
+			DailySignupCountType,
+		))
+
+		for loginIDType, count := range signupCountResult.CountByLoginID {
+			counts = append(counts, NewDailySignupWithLoginID(
+				appID,
+				count,
+				date,
+				loginIDType,
+			))
+		}
+
+		for provider, count := range signupCountResult.CountByOAuthProvider {
+			counts = append(counts, NewDailySignupWithOAuth(
+				appID,
+				count,
+				date,
+				provider,
+			))
+		}
+
+		if signupCountResult.AnonymousCount != 0 {
+			counts = append(counts, NewCount(
+				appID,
+				signupCountResult.AnonymousCount,
+				date,
+				DailySignupAnonymouslyCountType,
+			))
+		}
+		return nil
+	})
+	if err != nil {
+		return
+	}
+
 	return
 }
+
+func (c *CountCollector) QuerySignupCount(appID string, rangeFrom *time.Time, rangeTo *time.Time) (*SignupCountResult, error) {
+	var first uint64 = 100
+	var after model.PageCursor = ""
+
+	result := &SignupCountResult{
+		CountByLoginID:       map[string]int{},
+		CountByOAuthProvider: map[string]int{},
+	}
+	for {
+		events, lastCursor, err := c.QueryUserCreatedEvents(appID, rangeFrom, rangeTo, first, after)
+		if err != nil {
+			return nil, err
+		}
+
+		// Termination condition
+		if len(events) == 0 {
+			return result, nil
+		}
+
+		after = lastCursor
+		for _, e := range events {
```

Comment:
I am thinking whether we should perform the aggregation directly in the SQL query.

Reply:
After discussion, we will keep the original implementation. Since in the sql approach, will need to check all the login methods even for the apps that don't have much activities. 

May consider use the sql approach if there are apps have many activities in the future.